lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -64,38 +64,77 @@ private function loadAttributes( $userId ) {
 	 * @param string $userId
 	 * @param Attribute $attribute
 	 */
-	public function setAttribute( $userId, $attribute ) {
+	public function setAttribute( $userId, Attribute $attribute ) {
 		if ( $this->isAnonUser( $userId ) ) {
 			return;
 		}
 
-		$this->loadAttributes( $userId );
-		$this->setAttributeInService( $userId, $attribute );
-		$this->setAttributeInCache( $userId, $attribute );
+		if ( $this->attributeValueAlreadySet( $userId, $attribute ) ) {
+			return;
+		}
 
+		$this->setAttributeInCache( $userId, $attribute );
+		$this->setAttributeInService( $userId, $attribute );
```

Comment:
Any reason the order of the two lines above were switched?  It seems like you'd want to set it in the service first.  If the service call fails now, the cache will be invalid.  When flipped, if the cache call fails, the service is still the holder of truth.